### PR TITLE
Add `uuid` and `uuid[]` prop types (indexed with roaring bitmaps)

### LINKED ISSUE #2734: FR: Native UUID property type
UUIDs are common in the industry and if users already use UUIDs in their processes, they would have to be represented as strings inside Weaviate. Storing a UUID as a string takes around 36 bytes, whereas the information contained is just 16 bytes. So that's a lot of overhead. With a native `uuid` type, this prop could be stored and indexed more efficiently.

### PR DESCRIPTION
**Pull request by etiennedi**
### What's being changed:
* This adds `uuid` and `uuid[]` property types. When indexed in the inverted indexed, those are not indexed as string representation (36 bytes), but as a 128-bit (16 byte) number
* The filterable index is served using `RoaringSet` (i.e. roaring bitmaps)
* It is currently not possible to Aggregate `uuid` or `uuid[]` types
* It is currently not possible to sort by `uuid` or `uuid[]` types
* closes #2734 

### Usage Example
Screen Shot 2023-03-09 at 10 57 42 AM

### Implementation checklist
* [x] add type definitions
* [x] parse at ingest time
* [x] serve when retrieving objects in GraphQL and REST
* [x] index as `RoaringSet` bucket
* [x] parse filter when uuid specified as `valueString`
* [x] add e2e tests
* [x] add appropriate error message when trying to sort 

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Comment by jake-sager** [comment]
Epic!

**Comment by etiennedi** [comment]
@modi95 and @jake-sager if you want to try it out already, here's a [preview build](https://github.com/weaviate/weaviate/actions/runs/4373804008#summary-11878131598).

File: adapters/repos/db/inverted/objects_test.go [hunk 1]
```diff
@@ -57,6 +64,26 @@ func TestAnalyzeObject(t *testing.T) {
...
+			{
+				Name:         "idArray1",
+				DataType:     []string{"uuid[]"},
+				Tokenization: "field",
+			},
+			{
+				Name:         "idArray2",
+				DataType:     []string{"uuid[]"},
+				Tokenization: "field",
+			},
```
Comment (aliszka) [inline adapters/repos/db/inverted/objects_test.go:86]: is tokenization needed for uuids?

File: adapters/repos/db/filters_integration_test.go [hunk 2]
```diff
@@ -727,6 +747,16 @@ var carClass = &models.Class{
...
+		{
+			DataType:     []string{string(schema.DataTypeUUID)},
+			Name:         "manufacturerId",
+			Tokenization: "field",
+		},
+		{
+			DataType:     []string{string(schema.DataTypeUUIDArray)},
+			Name:         "availableAtDealerships",
+			Tokenization: "field",
+		},
```
Comment (aliszka) [inline adapters/repos/db/filters_integration_test.go:759]: is tokenization needed for uuids?

**Review by aliszka — APPROVED** [review]

File: adapters/repos/db/inverted/objects_test.go (hunk 1 above)
Comment (etiennedi) [reply adapters/repos/db/inverted/objects_test.go:86]: Good question. It shouldn't. It's mainly there from copy/pasting. Let me see what happens if I remove it.

Comment (etiennedi) [reply adapters/repos/db/inverted/objects_test.go]: You were correct, they are not needed. I've removed them in the latest commit.

**Pull request merged**